src/index.ts

```diff
@@ -0,0 +1,141 @@
+   */
+  private getHeaders(): Record<string, string> {
+    const headers: IHeaders = {
+      Authorization: this.clientKey,
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    };
+
+    if (this.etag !== '') {
+      headers['If-None-Match'] = this.etag;
+    }
+
+    return headers as unknown as Record<string, string>;
+  }
+
+  /**
+   * Asynchronously fetches toggle data from the feature toggle provider
+   * 
+   * If the request is successful and the response status is not 304 (Not Modified),
+   * it updates the instance's `etag` and `toggles` properties with the new data from the response.
+   * 
+   * The method returns a `FetchTogglesStatus` indicating whether the toggles were updated
+   * or remained unchanged. If the response is not successful (other than a 304 status), 
+   * an error is thrown indicating the failure to fetch toggles.
+   *
+   * @returns A promise that resolves to a `FetchTogglesStatus` enum, either `Updated`
+   * if new data was fetched or `Unchanged` if no updates were needed.
+   *
+   * @throws Throws an error if the request is not successful (other than a 304 status).
+   */
+  public async fetchToggles(): Promise<FetchTogglesStatus> {
+    const url = urlWithContextAsQuery(this.url, this.context);
+    const response = await fetch(url.toString(), {
+      method: 'GET',
+      cache: 'no-cache',
+      headers: this.getHeaders(),
+    });
+
+    if (!response.ok && response.status !== 304) {
+      throw new Error('Unable to fetch toggles.');
+    }
+
+    if (response.ok && response.status !== 304) {
+      const data = await response.json();
+
+      this.etag = response.headers.get('ETag') || '';
+      this.toggles = data.toggles as unknown as Toggle[];
+
+      return FetchTogglesStatus.Updated;
+    }
+
+    return FetchTogglesStatus.Unchanged;
+  }
+
+  /**
+   * Retrieves the toggles.
+   * 
+   * Returns the array of toggles if they have been previously downloaded.
+   * If the toggles are not yet downloaded, it throws an error.
+   *
+   * @returns An array of toggles if available.
+   * @throws {Error} If the toggles have not been downloaded.
+   */
+  public getToggles(): Toggle[] {
+    if (!this.toggles) {
+      throw new Error('Toggles not downloaded yet.');
+    }
+
+    return this.toggles;
+  }
+
+  /**
+   * Checks if a toggle with a given name is enabled.
+   * 
+   * This method iterates through the toggles and returns `true` if a toggle with the specified
+   * name is found and is enabled (or false if it's not).
+   *
+   * If the toggles with the given name is not found, it returns `false`.
+   *
+   * An error is thrown if the toggles have not been loaded.
+   *
+   * @param name - The name of the toggle to check.
+   * @returns `true` if the toggle is found and enabled, otherwise `false`.
+   * @throws {Error} If the toggles have not been downloaded.
+   */
+  public isEnabled(name: string): boolean {
```

Comment:
Does it make sense to add here a parameter with a default value in case the toggle is not found? Or do you prefer to keep the same as the library?

How are we handling default values, in case we prefer that it's enabled by default?

Reply:
If the toggle is deactivated for this specific user, it will not be found here, unleash never sends a toggle with `enabled: false`, so we can't add a default for the case that the toggle is not found

We could have a default for when the toggles are not downloaded yet, but I prefer to throw an error